### ISSUE TITLE
Log4J vulnerability handling

### ISSUE THREAD

**Issue by kevangelidis** [spinnaker/spinnaker#6575]
### Additional Details: N/A

**Comment by hivl** [comment]
Im really not sure, but i thought that spinnaker microservices are based on spring boot. I scanned images and didnt found any use of log4j-core.
https://spring.io/blog/2021/12/10/log4j2-vulnerability-and-spring-boot

**Comment by dogonthehorizon** [comment]
@hivl correct, Spinnaker in particular uses slf4j + logback as its logging implementation. Log4j is pulled in as a transitive dependency but isn't exposed to users in any capacity.

@kevangelidis can you help us understand what your concern is, as well as what a resolution on this issue looks like for you?

**Comment by sarkas52** [comment]
Hi we are using spinnaker 1.19.11 and hosted on K8 ,after Vulnerabilities scanning we find out that  spin-orca,spin-front50, spin-igor, spin-rosco,spin-clouddriver,spin-echo, spin-gate pods are using Log4J 2.12.1

**Comment by dogonthehorizon** [comment]
@sarkas52 can you clarify what you mean by using? I'll restate that log4j is pulled in as a transitive dependency but there is no codepath that would allow the RCE to be exploited.

**Comment by kevangelidis** [comment]
Hello @dogonthehorizon, thanks for the answer and for the clarifications. I understand what you mention, but I think that the Log4J vulnerability allows remote access execution, despite the fact that is used as a depedency. 

To give you an example, what was reported by apache is, one can send a request with some parameters or headers or json body, for example {create new user X}. In case this is logged from Log4J, it can be executed as admin. You can then imagine the effects that this can cause.

So to sum up, I think that you should release a new spinnaker versions that will use Log4J version 2.15 or greater. Of course, if you can investigate this issue on your side, I am willing to hear your feedback.

Thanks again and happy new year!

**Comment by jasonmcintosh** [comment]
@kevangelidis this isn't entirely correct.  First, I'll reference Apache's security release page: https://logging.apache.org/log4j/2.x/security.html

Additionally, Armory has published:
https://support.armory.io/support?id=kb_article&sysparm_article=KB0010516

* the vulnerability is only exercisable via a specific class that does JNDI handling.  This class exists in log4j-core which is NOT included.  The log4j-api & such libraries ARE included transitively via various projects.  it's always possible some FUTURE JAR file COULD add these, which is why there's a flag now in the core library preventing this.
* There ARE other vulnerabilities including some denial of service attack - the apache page has a list of them.  BUT library vulnerabilities again do NOT mean that projects which include the library are implicitly vulnerable.  It's always possible a code path could trigger behavior, but not in this case.
* To trigger this behavior you ALSO have to be LOGGING attacker payload in some fashion.  E.g. "Headers or json body" would have to be LOGGED.  And again, since we're not logging via log4j, but logback, no impact.  AND even then, to trigger the attack even if log4j is used, you'd need that library (which without, log4j doesn't really work anyways).

SO yeah :) Spinnaker is safe.  NOW you could be impacted if you were using sidecars or some OTHER system and other agent processes.  Just because spinnaker is safe, doesn't mean other systems are safe.

**Comment by kevangelidis** [comment]
Aha, ok. Got it Jason. 
Thanks a lot for the prompt reply. Really appreciate it.
I think we can close this issue. Wish you happy new year!

**Comment by jasonmcintosh** [comment]
NP - happy to answer other questions on this if you have any ;). ALSO watch out - we JUST published a CRITICAL cve for spinnaker itself.  https://github.com/spinnaker/spinnaker/security/advisories
